mink/tasks/equality_constraint_task.py

```diff
@@ -0,0 +1,186 @@
+    identified by name or ID.
+
+    Attributes:
+        equalities: ID or name of the equality constraints to regulate. If not provided,
+            the task will regulate all equality constraints in the model.
+        cost: Cost vector for the equality constraint task. Either a scalar, in which
+            case the same cost is applied to all constraints, or a vector of shape
+            ``(neq,)``, where ``neq`` is the number of equality constraints in the
+            model.
+
+    Raises:
+        InvalidConstraint: If a specified equality constraint name or ID is not found.
+        TaskDefinitionError: If no equality constraints are found or if cost parameters
+            have invalid shape or values.
+
+    Example:
+
+    .. code-block:: python
+
+        # Regulate all equality constraints with the same cost.
+        eq_task = EqualityConstraintTask(model, cost=1.0)
+
+        # Regulate specific equality constraints with different costs.
+        eq_task = EqualityConstraintTask(
+            model,
+            cost=[1.0, 0.5],
+            equalities=["connect_right", "connect_left"]
+        )
+    """
+
+    def __init__(
+        self,
+        model: mujoco.MjModel,
+        cost: npt.ArrayLike,
+        equalities: Optional[Sequence[int | str]] = None,
+        gain: float = 1.0,
+        lm_damping: float = 0.0,
+    ):
+        self._mask: np.ndarray | None = None  # Active equality constraint mask.
+        self._neq_active: int | None = None  # Number of active equality constraints.
+
+        eq_ids: list[int] = []
+        if equalities is not None:
+            for eq_id_or_name in equalities:
+                eq_id: int
+                if isinstance(eq_id_or_name, str):
+                    eq_id = mujoco.mj_name2id(
+                        model, mujoco.mjtObj.mjOBJ_EQUALITY, eq_id_or_name
+                    )
+                    if eq_id == -1:
+                        raise InvalidConstraint(
+                            f"Equality constraint '{eq_id_or_name}' not found."
+                        )
+                else:
+                    eq_id = eq_id_or_name
+                    if eq_id < 0 or eq_id >= model.neq:
+                        raise InvalidConstraint(
+                            f"Equality constraint index {eq_id} out of range."
+                            f"Must be in range [0, {model.neq})."
+                        )
+                eq_ids.append(eq_id)
+        else:
+            eq_ids = list(range(model.neq))
+            logging.warning("Regulating %d equality constraints", len(eq_ids))
+        if len(eq_ids) == 0:
+            raise TaskDefinitionError(
+                f"{self.__class__.__name__} no equality constraints found in this "
+                "model."
+            )
+        self._eq_ids = np.array(eq_ids)
+        self._eq_types = model.eq_type[self._eq_ids]
+        self._neq_total = len(self._eq_ids)  # Total number of equality constraints.
```

Comment:
Maybe overkill, but somewhere around here might be good to check `mjModel.eq_active0` ?
The user is allowed to define equalities but tell MuJoCo to disable them...

Reply:
I decided to throw an error if the user asks me to regulate a constraint but its entry in `eq_active0` is False. The reason is that it makes it a bit undefined what to do with the cost: if the user gave me 2 costs for 2 constraints and one of them is inactive, do I ignore the cost and warn the user? Do I throw an error? I opted for clarity instead.